elitzur-core/src/main/scala/com/spotify/elitzur/validators/DynamicRecordValidator.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2021 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.spotify.elitzur.validators
+
+import com.spotify.elitzur.MetricsReporter
+
+
+@SuppressWarnings(Array("org.wartremover.warts.Var"))
+final private[elitzur] case class DynamicRecordValidator(validators: Array[Validator[Any]],
+                                                         labels: Array[String])
+                                                        (implicit reporter: MetricsReporter)
+  extends Validator[Seq[Any]] {
+
+  override def validateRecord(a: PreValidation[Seq[Any]],
+                              path: String = "",
+                              outermostClassName: Option[String] = None,
```

Comment:
Maybe we should add a light wrapper function for this that requires outermostClassName (since that errors below if its empty) and doesn't require path (if nested records are already flattened here). 

 It seems we won't be able to re-use the existing ValidatorDoFn since that doesn't set outermostClassName anyway

Reply:
Gonna leave the specifics of this to figure out in the pipelining PR since I don't want to create something that might not get used if that impl changes.